erasure-createfile.go

```diff
@@ -44,10 +44,15 @@ func erasureCreateFile(disks []StorageAPI, volume, path string, reader io.Reader
 			return 0, nil, rErr
 		}
 		if rErr == io.EOF {
+			// Should return IncompleteBody when reader contains
+			// fewer than specified bytes
+			if bytesWritten < size {
+				return bytesWritten, nil, io.ErrShortWrite
```

Comment:
@harshavardhana Nothing stops us from handling it at the object layer. I shall update the PR handling it at the object layer for XL.